metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/graphite/GraphiteQueryServiceTest.java

```diff
@@ -169,4 +169,37 @@ public void testGetMetricQueryResultEmpty() {
                 metricDataResult.getMetricDefinition().getMeta());
         }
 
+    @Test
+    public void testGetMetricQueryResultNullDatapoint() {
+        val uri = "samplegraphitehosturi/render?until=now&format=json&target=sumSeries(a.b.c)&from=-30s";
+        val testDatapoint = "[null,1568255056]";
+        JSONArray sampleJsonGraphite = new JSONArray();
+        JSONObject sampleJsonGraphiteResult = new JSONObject();
+        JSONArray testDatapoints = new JSONArray();
+        JSONObject testTags = new JSONObject();
+        testTags.put("aggregatedBy", "sum");
+        testTags.put("name", "sumSeries(a.b.c)");
+        testDatapoints.put(0, testDatapoint);
+        sampleJsonGraphiteResult.put("datapoints", testDatapoints);
+        sampleJsonGraphiteResult.put("target", "sumSeries(a.b.c)");
+        sampleJsonGraphiteResult.put("tags", testTags);
+        sampleJsonGraphite.put(0, sampleJsonGraphiteResult);
+        Content mockGraphiteResult = new Content(sampleJsonGraphite.toString().getBytes(), ContentType.APPLICATION_JSON);
+        try {
+            when(httpClientWrapper.get(uri)).thenReturn(mockGraphiteResult);
+        }
+        catch (Exception e) {
+            log.error("{}", e);
+        }
```

Comment:
After logging this exception, control continues flowing past the try/catch. Is that really the desired behavior?

I understand that it will not actually throw an exception here. But what I would suggest is just remove the try/catch and make the test method itself throw an `Exception`. That way if you get an unexpected exception, the test fails.

Reply:
Updated.